src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/actions/GenerateTestsActionMethod.kt

```diff
@@ -19,50 +16,30 @@ import nl.tudelft.ewi.se.ciselab.testgenie.helpers.generateMethodDescriptor
  *   getting the information about the selected method and passing it to (EvoSuite) Runner.
  */
 class GenerateTestsActionMethod : AnAction() {
-    private val log = Logger.getInstance(this.javaClass)
-
     /**
-     * Performs test generation for a method when the action is invoked.
+     * Creates and calls (EvoSuite) Runner to generate tests for a method when the action is invoked.
      *
-     * @param e an action event that contains useful information
+     * @param e an action event that contains useful information and corresponds to the action invoked by the user
      */
     override fun actionPerformed(e: AnActionEvent) {
-        val project: Project = e.project ?: return
+        val evoSuiteRunner: Runner = createEvoSuiteRunner(e) ?: return
 
         val psiFile: PsiFile = e.dataContext.getData(CommonDataKeys.PSI_FILE) ?: return
         val caret: Caret = e.dataContext.getData(CommonDataKeys.CARET)?.caretModel?.primaryCaret ?: return
-        val vFile = e.dataContext.getData(CommonDataKeys.VIRTUAL_FILE) ?: return
-        val fileName = vFile.presentableUrl
-        val modificationStamp = vFile.modificationStamp
 
         val psiMethod: PsiMethod = getSurroundingMethod(psiFile, caret) ?: return
-        val containingClass: PsiClass = psiMethod.containingClass ?: return
-
-        val classFQN = containingClass.qualifiedName ?: return
         val methodDescriptor = generateMethodDescriptor(psiMethod)
 
-        val projectPath: String = ProjectRootManager.getInstance(project).contentRoots.first().path
-        val projectClassPath = "$projectPath/target/classes/"
-
-        log.info("Generating tests for project $projectPath with classpath $projectClassPath")
-
-        log.info("Selected class is $classFQN, method is $methodDescriptor")
+        Logger.getInstance("GenerateTestsUtils").info("Selected method is $methodDescriptor")
```

Comment:
You should use the logger you removed above.

Reply:
Why? I don't think it is necessary to store it as a separate private variable now that it prints only one line. At least to me it seems like an overhead. If i understand correctly what you mean

Reply:
I meant the name of the logger - this logger is named after GenerateTestsUtils but you use it in the GenerateTestsActionMethod.

(Also, creating a private variable won't create any overhead, don't worry about that)

Reply:
Is it better to create one or not? I would not, personally

Reply:
I think it would be better to create a private variable since it allows for easier extensibility when you want to modify this class and add another log message.

Reply:
Done